core/src/main/scala/org/apache/spark/api/python/PythonRDD.scala

```diff
@@ -866,11 +866,14 @@ class BytesToString extends org.apache.spark.api.java.function.Function[Array[By
 }
 
 /**
- * Internal class that acts as an `AccumulatorParam` for Python accumulators. Inside, it
+ * Internal class that acts as an `AccumulatorV2` for Python accumulators. Inside, it
  * collects a list of pickled strings that we pass to Python through a socket.
  */
-private class PythonAccumulatorParam(@transient private val serverHost: String, serverPort: Int)
-  extends AccumulatorParam[JList[Array[Byte]]] {
+private[spark] class PythonAccumulatorV2(@transient private val serverHost: String, serverPort: Int)
+  extends AccumulatorV2[JList[Array[Byte]], JList[Array[Byte]]] {
```

Comment:
So not really - the type signatures don't quite match. We do an add of a List of values on the worker threads but CollectionAccumulator expects the add to be of the type being accumulated into the list. We could use merge or add a separate add function, but it would be the only place where merge is called on the workers and I'm hesitant to do that since at that point we really aren't looking much like the accumulators.